src/main/java/com/amazon/opendistroforelasticsearch/sql/query/ESActionFactory.java

```diff
@@ -91,6 +91,13 @@ else if(isJoin(sqlExpr,sql)){
                 else {
                     sqlExpr.accept(new TermFieldRewriter(client));
                     Select select = new SqlParser().parseSelect(sqlExpr);
+                    if (select.isSelectAll() && !select.getFields().isEmpty()) {
```

Comment:
How about we move this new added logic to `SqlParser.findSelect()` to fail fast?